catalog/data_metastores.go

```diff
@@ -14,12 +14,12 @@ func DataSourceMetastores() *schema.Resource {
 		Ids map[string]string `json:"ids,omitempty" tf:"computed"`
 	}
 	return common.AccountData(func(ctx context.Context, data *metastoresData, acc *databricks.AccountClient) error {
-		metastores, err := acc.Metastores.List(ctx)
+		metastores, err := acc.Metastores.ListAll(ctx)
```

Comment:
Just quickly changes this to have the code compile with the new SDK. Not the correct change (test fails). I will fix it.